Add link to latest job in each scenario

okurz opened the pull request:
Based on 'latest' query route each job can now link to the
newest incarnation within in the same scenario.

Add 'latest' query route
    
Should always refer to most recent job for the specified scenario.
    
* have the same link for test development, i.e. if one retriggers tests, the
person has to always update the URL. If there would be a static URL even the
browser can be instructed to reload the page automatically
    
* for linking to the always current execution of the last job within one
scenario, e.g. to respond faster to the standard question in bug reports "does
this bug still happen?"

As suggested in #796

Example screenshot:
![openqa_link_to_latest](https://cloud.githubusercontent.com/assets/1693432/17368118/0f92b70e-5993-11e6-9afa-a04210492a31.png)

coolo commented on lib/OpenQA/Schema/Result/Jobs.pm:
```diff
@@ -220,6 +224,14 @@ sub name {
+sub scenario_hash {
+    my ($self) = @_;
+    my %scenario = map { lc $_ => $self->get_column($_) } SCENARIO_KEYS, 'MACHINE';
+    use Data::Dumper;
```
this looks like debug

okurz replied:
done

coolo commented:
If you want nicer URLs you can also have /tests/latest/sle/Server-DVD/... - i.e. have required URL components for parameters. Not so sure if tests/latest?distri=sle makes so much sense

okurz commented:
updated:
* removed debug output
* using two constant lists for scenario keys w and w/o machine

`tests/latest?distri=sle` could be like Google's "I'm feeling lucky" button ;-) `tests/latest?distri=my_debug_distri` might make sense or current `tests/latest?distri=windows` on o3. I don't think we have to forbid the use of such cases. Required URL components would make sense if they are really required. Maybe we can also add this additionally later.

coolo commented:
"Clone of 490580 Latest in same scenario" looks really, really strange.

We already talked about adding more recent results into the same tab where we have 'previous results' now - in there a link to the latest is to be expected. Somewhere in the middle of nowhere in the info_box not.

I'm fine with merging the rest, so remove the link for now

okurz commented:
> I'm fine with merging the rest, so remove the link for now

that would be #815 then

okurz commented:
updated:
* rebased after merge of latest_query
* added example screenshot to description

okurz commented:
obsoleted by #836
